builtin/logical/ssh/path_roles.go

```diff
@@ -512,6 +525,22 @@ func (b *backend) createCARole(allowedUsers, defaultUser, signer string, data *f
 		NotBeforeDuration:         time.Duration(data.Get("not_before_duration").(int)) * time.Second,
 	}
 
+	// HACK: I don't like this check, refactor this later
+	if notBeforeAbsoluteRaw, ok := data.GetOk("not_before_absolute"); ok {
+		switch v := notBeforeAbsoluteRaw.(type) {
```

Comment:
I think you can just do:

```
role.NotBeforeAbsolute = notBeforeAbsoluteRaw.(time.Time)
```

and it will be safe to type assert based on the above.

Reply:
So, `GetOk(...)` has different semantics than `Get(...)` in that if the value isn't provided, `GetOk` will return `false`, so we'll skip the assignment. We don't have a default value, which is fine, but if we call `Get(...)` and assign to `time.Time` or `*time.Time`, we'll fail: https://go.dev/play/p/TGU0JR5GIRn 

So, I think what this needs to be is, outside of the above role entry, just like you've done here, we'd do:

```go
notBeforeRaw := data.Get("not_before")
if notBeforeRaw != nil {
  role.NotBefore = notBeforeRaw.(time.Time)
}
```

(and because we're creating `role` fresh above, we'll have it set to the default zero value)